later/lib/domain/remap/record_remapper_impl.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:later/database/models/record_model.dart';
+import 'package:later/domain/models/record_model.dart';
+import 'package:later/domain/remap/record_remapper.dart';
+
+class RecordRemapperImpl extends RecordRemapper {
```

Comment:
As `RecordRemapper` is an interface, prefer using the `implements` keyword instead of `extends`. There's no difference in this case, but it generally looks nicer (`Impl` class _implements_ an interface)

```suggestion
class RecordRemapperImpl implements RecordRemapper {
```